lib/plugins/aws/package/compile/events/alb/lib/validateAuthorizers.test.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const expect = require('chai').expect;
+const AwsCompileAlbEvents = require('../index');
+const Serverless = require('../../../../../../../Serverless');
+const AwsProvider = require('../../../../../provider/awsProvider');
+
+describe('#validate()', () => {
+  let awsCompileAlbEvents;
+
+  beforeEach(() => {
+    const serverless = new Serverless();
+    serverless.setProvider('aws', new AwsProvider(serverless));
+    serverless.service.service = 'some-service';
+    serverless.service.provider.compiledCloudFormationTemplate = { Resources: {} };
+    awsCompileAlbEvents = new AwsCompileAlbEvents(serverless);
+  });
+
+  describe('#validateAuthorizers()', () => {
+    let baseAlbConfig;
+
+    beforeEach(() => {
+      baseAlbConfig = {
+        authenticateCognito: {
```

Comment:
Let's not add tests that confirm on validation (as that will eventually be handled by config schema).

Concerning tests ideally it would be good to have some integration tests that confirm on basic setup of `cognito` and `oidc` authorizers. Integration tests are placed here: https://github.com/serverless/serverless/tree/962506b4356545870e18d570756240e602b5f541/tests/integration-all (it appears we don't have any for ALB  yet). We already also have cognito authorizer test at HTTP API, that could serve as inspiration: https://github.com/serverless/serverless/blob/962506b4356545870e18d570756240e602b5f541/tests/integration-all/http-api/tests.js#L166-L187

If for some reason it's difficult to setup reliable integration test, feel free to skip that.

Otherwise all new tests we introduce are made with `runServerless` util, as it's the only reliable way to test the outcome, and such test doesn't depend as much on nature of internal implementation (which should rather be treated as a black box).

You may check lambda destination tests as example: https://github.com/serverless/serverless/blob/962506b4356545870e18d570756240e602b5f541/lib/plugins/aws/package/compile/functions/index.test.js#L2624-L2694

I've also provided some explanation in that comment: https://github.com/serverless/serverless/pull/7335#discussion_r389322881

(sorry for not having yet proper documentation on that)

Reply:
Integration tests seem to be quite a challenge. Like when i try run them locally I get a load of errors
![Screen Shot 2020-03-26 at 11 49 55](https://user-images.githubusercontent.com/6001129/77643834-f155ee80-6f57-11ea-9b7a-f5c9a9883b58.png)
What i need AWS creds?
Is it ok if I omit integration tests for this PR, then later on when I have more time I can come back to this? Just that a project I'm working on needs this functionality asap. Once I have more time I can come back to this and in a separate PR add integration tests (and hopefully by then they'll be better documentation)

Reply:
Integration tests, test directly the integration against AWS service. Hence access to some AWS account is needed.

Technically it's the only reliable way to issue e2e test for given functionality.

Still if you find challenging to setup one feel free to skip it.

Reply:
Makes sense. But I already have an aws account configured and the creds set in ~/.aws/credentials. Where does it expect to get the creds from?

Reply:
Can I skip for now, and I can come back to this later on. But do you think the work done so far is merge ready?

Reply:
> But I already have an aws account configured and the creds set in ~/.aws/credentials.

Tests are run in isolated environment, so also HOME directory points temp dir, that's why your `~/aws/credentials` in not seen.
To have it working you need to ensure `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars

Reply:
> Can I skip for now, and I can come back to this later on. But do you think the work done so far is merge ready?

? @medikoo

Reply:
Very little to work from as there are no load balancer integration tests at all. Like where do I even add the cloud formation scripts to create a load balancer, listener and target group?

Reply:
@shadrech I'll try to review in next days. Unfortunately my priorities were shifted (temporarily) now, that's why there's such delay. Sorry for delay!

Reply:
All good